src/js/sync.js

```diff
@@ -0,0 +1,93 @@
+var SailthruGigya = {
+
+  options : {},
+  profile : {},
+
+  init : function(options) {
+
+    if (!window.Sailthru) {
+      console.log('Sailthru JS is not loaded.Ensure you have included the JS on page');
+      return;
+    }
+
+    if (gigya) {
+        this.options = options;
+        gigya.socialize.addEventHandlers({
+          onLogin:this.syncProfile
+       });
+    } 
+  },
+
+  syncProfile : function(eventObj) {
+
+      // Allow customer to send this to server side too for additional processing.  
+      if (SailthruGigya.options.callback_url) {
+        SailthruGigya.processCallback(eventObj, SailthruGigya.options.callback_url);
+      }
+
+      var userKey = '';
+      var vars = {};
+      var lists = {};
+      var params = {};
+
+
+      if (eventObj.user.email !=='' || typeof eventObj.user.email != 'undefined') {
```

Comment:
This should be 
`
typeof eventObj.user.email == 'undefined'
`

Reply:
Can this be rewritten so that it exits if *not* this case? Means less code indentation, and clearer flow of logic. 

Easier to read an `if` for an exit/stop than to go into something.